osu.Server.Spectator/S3.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Amazon;
+using Amazon.Runtime;
+using Amazon.S3;
+using Amazon.S3.Model;
+
+namespace osu.Server.Spectator
+{
+    public static class S3
+    {
+        public static byte[]? Retrieve(string bucket, string key, RegionEndpoint? endpoint = null)
```

Comment:
Bit peculiar that this is the only method that accepts a `RegionEndpoint` and the rest just defaults to `USWest1`.

Reply:
I’ll remove the other methods, this entire class is a carbon copy from another project that needed these.